### PR TITLE
Wrap Crypto, DateTime, Password, X509, NoModule, and HttpClient built-in functions

### PR DESCRIPTION
**Pull request by OceanOak**
No changelog

This PR creates wrapper functions around Builtin Crypto, DateTime, Password, X509, NoModule, and HttpClient functions

### CODE REVIEW

File: packages/darklang/stdlib/json.dark [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Darklang =
+  module Stdlib =
+    module Json =
```
Comment (OceanOak) [inline packages/darklang/stdlib/json.dark:3]: TODO: recheck this module, it doesn't work

File: packages/darklang/stdlib/noModule.dark [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    /// Returns true if the two value are equal
+    let equals (a: 'a) (b: 'a) : Bool = Builtin.equals a b
+
+
+    /// Returns true if the two value are not equal
+    let notEquals (a: 'a) (b: 'a) : Bool = Builtin.notEquals a b
+
+
+    /// Unwrap an Option or Result, returning the value or a DError if None
+    let unwrap (value: 'a) : 'a = Builtin.unwrap value
```
Comment (OceanOak) [inline packages/darklang/stdlib/noModule.dark:12]: TODO: check if this can be rewritten in dark

Comment (pbiggar) [reply packages/darklang/stdlib/noModule.dark:12]: I'm wondering if we should even expose this, or if it should only exist as a `!` once we have darklang syntax for it?

File: packages/darklang/stdlib/httpclient.dark [hunk 3]
```diff
@@ -2,46 +2,19 @@ module Darklang =
...
-        ("content-type", "text/plain; charset=utf-8")
-
-      /// Returns a header <type (String*String))> with {{'content-type'}} set for html requests
-      let htmlContentType () : (String * String) =
-        ("content-type", "text/html; charset=utf-8")
\ No newline at end of file
+      /// The response from a HTTP request
+      type Response =
+        { statusCode: Int
+          headers: List<Tuple<String, String>>
```
Comment (pbiggar) [inline packages/darklang/stdlib/httpclient.dark:8]: `Tuple` is odd here. It should be `String * String`

File: packages/darklang/stdlib/httpclientAuth.dark [hunk 4]
```diff
@@ -0,0 +1,47 @@
+module Darklang =
+  module Stdlib =
+    module HttpClientAuth =
```
Comment (pbiggar) [inline packages/darklang/stdlib/httpclientAuth.dark:3]: These shouldn't be in a new module, they should be in httpclient.

File: packages/darklang/stdlib/noModule.dark (hunk 2 above)
Comment (pbiggar) [reply packages/darklang/stdlib/noModule.dark:12]: It can't be rewritten in Dark though

**Comment by pbiggar** [comment]
Looking good.

**Comment by pbiggar** [comment]
Nice, thanks!

**Pull request merged**